src/4.3_to_5.0/migrateKpi.ts

```diff
@@ -210,20 +210,27 @@ export function migrateKpi(
     }),
   };
 
+  const migratedTitles: KpiWidgetState["titles"] = {};
   try {
     if (legacyMdx) {
       // Migrate manually entered KPI titles.
-      const migratedTitles = getMigratedKpiTitles(legacyKpiState, {
+      accumulateMigratedKpiTitles(legacyKpiState, {
         cube,
         mapping,
         legacyMdx,
+        migratedTitles,
       });
       if (migratedTitles && Object.keys(migratedTitles).length > 0) {
         migratedWidgetState.titles = migratedTitles;
       }
     }
   } catch (error) {
     // Migrating the KPI titles is a best effort.
+    // If there is an error while migrating one of the titles, the successfully migrated titles are still persisted.
+    if (migratedTitles && Object.keys(migratedTitles).length > 0) {
+      migratedWidgetState.titles = migratedTitles;
```

Comment:
If we encounter an error, this means that we will save the already migrated KPI titles, but we will still lose all the ones that weren't yet reached in the iteration, right?

A more ideal behavior would be to even keep doing the iteration and keep accumulating and saving the rest of KPI titles.

Reply:
Actually, the behavior is already as you suggest, internally `accumulatedMigratedKpiTitles` "swallows" any error until all the titles have attempted to be migrated. Once all titles have attempted to be migrated, it throws an error to mark the widget "partially" migrated.

Reply:
Got it now, thank you 👍